lib/matplotlib/__init__.py

```diff
@@ -496,7 +496,12 @@ def matplotlib_fname():
     """
 
     def gen_candidates():
-        yield os.path.join(os.getcwd(), 'matplotlibrc')
+        try:
+            # if the user has ended up with a non-existent directory
+            # as the current working directory do not fail to import.
+            yield os.path.join(os.getcwd(), 'matplotlibrc')
+        except FileNotFoundError:
+            pass
```

Comment:
I think I'd feel better if a log message was emitted. People may be wondering why their matplotlibrc file isn't getting picked up.

Reply:
what matplotlibrc? $cwd already doesn't exist, so $cwd/matplotlibrc certainly doesn't either?

Reply:
Also, it may be easier to just return a relative path here (`return "matplotlibrc"`) and then absolutize it at exit (`if os.path.exists(fname) and not os.path.isdir(fname): return os.path.abspath(fname)`), which would also absolutize something set from $MATPLOTLIBRC (well, I'm not sure we ever detailed the semantics of relative-path $MATPLOTLIBRC...)

Reply:
@anntzer, I am referring to a hypothetical matplotlibrc file that could have been in the `cwd()`, but is no longer there because the user didn't realize that they are running matplotlib in a directory that has been removed (an extension on the scenario described by @tacaswell). A log message here could provide a useful breadcrumb for a user to come to this realization. The use of relative paths as you'd suggested would actually paper over this problem and make it harder for a user to realize what is going on.

Reply:
I'm not sure I follow.  The same case would apply to a hypothetical matplotlibrc that's in ~/.config/matplotlib/matplotlibrc and that somehow got deleted before you import matplotlib; we don't warn in that case either (i.e. if there's no ~/.config/matplotlib/matplotlibrc, or ~/.config/matplotlib for that matter).

Reply:
If we go this route I don't think we need to abspath it. These paths go to `open` in very short order we can let it take care of relative paths.

@WeatherGod I'm with @anntzer here, unless we warn when we don't find a local matplotlibrc (but with the folder existing) I don't think we should warn when we do not find a local matplotlibrc (because $cwd does not exist).